dango/account/factory/src/execute.rs

```diff
@@ -199,24 +224,25 @@ fn onboard_new_user(
     ACCOUNTS.save(storage, address, &account)?;
     ACCOUNTS_BY_USER.insert(storage, (&username, address))?;
 
-    // Create the message to instantiate this account.
-    let msg = Message::instantiate(
-        code_hash,
-        &account::spot::InstantiateMsg { minimum_deposit },
-        salt,
-        Some(format!("dango/account/{}/{}", AccountType::Spot, index)),
-        Some(factory),
-        Coins::default(),
-    )?;
-
-    // Create the event indicating a new user has registered.
-    let event = UserRegistered {
-        username,
-        address,
-        key,
-    };
-
-    Ok((msg, event))
+    Ok((
+        Message::instantiate(
+            code_hash,
+            &account::spot::InstantiateMsg { minimum_deposit },
+            salt,
+            Some(format!("dango/account/{}/{}", AccountType::Spot, index)),
+            Some(factory),
+            Coins::default(),
+        )?,
+        UserRegistered {
+            username,
+            key,
+            key_hash,
+        },
+        AccountRegistered {
+            address,
+            params: account.params,
+        },
+    ))
```

Comment:
@Rhaki Previously you only emit a single `UserRegistered` event here. However, we're actually doing two things here: creating the user and creating the user's first, so there should be an event for each of these two actions. Now this function returns two events.

Reply:
@Rhaki Hmm but this impacts your bot, right? Cause now you need to mark the "open a subaccount" quest as complete only **the second time** you observe an `AccountRegistered` event for a given username.